src/Giver.sol

```diff
@@ -0,0 +1,144 @@
+    constructor(AddressDriver addressDriver_) {
+        addressDriver = addressDriver_;
+        _drips = addressDriver.drips();
+        _maxTotalBalance = _drips.MAX_TOTAL_BALANCE();
+
+        address nativeTokenWrapper_;
+        if (block.chainid == 1 /* Mainnet */ ) {
+            nativeTokenWrapper_ = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+        } else if (block.chainid == 5 /* Goerli */ ) {
+            nativeTokenWrapper_ = 0xB4FBF271143F4FBf7B91A5ded31805e42b2208d6;
+        } else if (block.chainid == 11155111 /* Sepolia */ ) {
+            nativeTokenWrapper_ = 0xfFf9976782d46CC05630D1f6eBAb18b2324d6B14;
+        } else {
+            nativeTokenWrapper_ = address(bytes20("native token wrapper"));
+        }
+        nativeTokenWrapper = IERC20(nativeTokenWrapper_);
+    }
+
+    /// @notice Calculate the address of the `Giver` assigned to the account ID.
+    /// The `Giver` may not be deployed yet, but the tokens sent
+    /// to its address will be `give`n when `give` is called.
+    /// @param accountId The ID of the account to which the `Giver` is assigned.
+    /// @return giver_ The address of the `Giver`.
+    function giver(uint256 accountId) public view returns (address giver_) {
+        return _giver(accountId, address(this));
+    }
+
+    /// @notice Calculate the address of the `Giver` assigned to the account ID.
+    /// @param accountId The ID of the account to which the `Giver` is assigned.
+    /// @param deployer The address of the deployer of the `Giver` and its logic.
+    /// @return giver_ The address of the `Giver`.
+    function _giver(uint256 accountId, address deployer) internal pure returns (address giver_) {
+        return
+            Clones.predictDeterministicAddress(_giverLogic(deployer), bytes32(accountId), deployer);
+    }
+
+    /// @notice Calculate the address of the logic that is cloned for each `Giver`.
+    /// @param deployer The address of the deployer of the `Giver` logic.
+    /// @param giverLogic The address of the `Giver` logic.
+    function _giverLogic(address deployer) internal pure returns (address giverLogic) {
+        // The address is calculated assuming that the logic is the first contract
+        // deployed by the instance of `GiversRegistry` using plain `CREATE`.
+        bytes32 hash = keccak256(bytes.concat(hex"D694", bytes20(deployer), hex"01"));
+        return address(uint160(uint256(hash)));
+    }
+
+    /// @notice `give` to the account all the tokens held by the `Giver` assigned to that account.
+    /// @param accountId The ID of the account to `give` tokens to.
+    /// @param erc20 The token to `give` to the account.
+    /// If it's the zero address, `Giver` wraps all the native tokens it holds using
+    /// `nativeTokenWrapper`, and then `give`s to the account all the wrapped tokens it holds.
+    /// @param amt The amount of tokens that were `give`n.
+    function give(uint256 accountId, IERC20 erc20) public whenNotPaused returns (uint256 amt) {
```

Comment:
This function is not callable on a non-proxied instance of `GiversRegistry` because it's `whenNotPaused` and it's always paused on deployment, see https://github.com/drips-network/contracts/blob/main/src/Managed.sol#L79C22-L79C22. 

An init function doing just `new Giver();` sounds like a smart move, why waste gas on checking if the giver logic is there?

Reply:
Added `initialize()`